elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -336,17 +352,16 @@ private Path buildPath(Type rootEntityType, String selector) {
             return new Path(path);
         }
 
-        private void parseArguments(String attributeName, Set<Argument> arguments) {
-            Matcher matcher = FILTER_ARGUMENTS_PATTERN.matcher(attributeName);
+        private void parseArguments(String argsString, Set<Argument> arguments) throws UnsupportedEncodingException {
+            if (argsString == null || argsString.isEmpty()) {
+                return;
+            }
+
+            Matcher matcher = Pattern.compile("\\[(\\w+?):([^\\]]+?)\\]").matcher(argsString);
```

Comment:
Can we use the original regex here?

Reply:
it doesn't work here  as the number of `[]` is unknown